.github/workflows/tests_and_cov.yml

```diff
@@ -14,21 +14,19 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Set up Python 3.12
+      - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: 3.12
+          python-version-file: ".python-version"
```

Comment:
Isn't it easier to just set a version here? Is it used only in the actions or also when running the app via uv?

Reply:
It's always used when calling uv, and also potentially by other tools - its another file, but I think it's pretty ok compared to search-and-replace 3.12 later when its time to change it.